File: 2211-k-radius-subarray-averages/2211-k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
         int n = nums.size(),i;
        long long int s=0;
        vector<int> ans(n,-1);
        for(i=0;i<n;i++){
            if(i<2*k){
                s+= nums[i];
            }else{
                s+=nums[i];
                ans[i-k] = s/(2*k+1);
                s -= nums[i-(2*k)];
            }
        }
        return ans;
    }
};

File: 2211-k-radius-subarray-averages/2211-k-radius-subarray-averages.java
class Solution {
    public int[] getAverages(int[] nums, int k) {
        int n=nums.length;
        int [] ans=new int[n];
        Arrays.fill(ans ,-1);
        
        if(k==0){
            return nums;

        }

        if(n < 2*k+1){
            return ans;

        }

        int left=0;
        int right=2*k;
        long sum=0;
        int i=k;
        for(int j=left; j<=right; j++){
            sum=sum+nums[j];
        }
        int avg=(int)(sum/(2*k+1));
        ans[i]=avg;
        i++;
        right++;
        
        while(right<n){
            int leftsidewindow=nums[left];
            int rightsidewindow=nums[right];
            sum=sum-leftsidewindow+rightsidewindow;

            ans[i]=(int)(sum/(2*k+1));
            i++;
            right++;
            left++;
        }
        

        
        return ans;
    }
}